reference/api/batches.mdx

```diff
@@ -0,0 +1,250 @@
+---
+title: Batches — Meilisearch API reference
+description: The /batches route allows you to monitor how Meilisearch is grouping and processing asynchronous operations.
+---
+
+# Batches
+
+The `/batches` route gives information about the progress of batches of [asynchronous operations](/learn/async/asynchronous_operations).
+
+## Batch object
+
+```json
+{
+  "uid": 0,
+  "details": {
+    "receivedDocuments": 6,
+    "indexedDocuments": 6
+  },
+  "stats": {
+    "totalNbTasks": 1,
+    "status": {
+      "succeeded": 1
+    },
+    "types": {
+      "documentAdditionOrUpdate": 1
+    },
+    "indexUids": {
+      "INDEX_NAME": 1
+    }
+  },
+  "duration": "PT0.250518S",
+  "startedAt": "2024-12-10T15:20:30.18182Z",
+  "finishedAt": "2024-12-10T15:20:30.432338Z",
+  "progress": {
+    "steps": [
+      { 
+        "currentStep": "extracting words",
+        "finished": 2,
+        "total": 9,
+      },
+      {
+        "currentStep": "document",
+        "finished": 30546,
+        "total": 31944,
+      }
+    ],
+    "percentage": 32.8471
+  }
+}
+```
+
+### `uid`
+
+**Type**: Integer<br />
+**Description**: Unique sequential identifier of the batch. Starts at `0` and increases by one for every new patch.
+
+### `details`
+
+**Type**: Object<br />
+**Description**: Basic information on the types tasks in a batch. Consult the [task object reference](/reference/api/tasks#details) for an exhaustive list of possible values.
+
+### `progress`
+
+**Type**: Object<br />
+**Description**: Object containing two fields: `steps` and `percentage`. Once Meilisearch has fully processed a batch, its `progress` is set to `null`.
+
+#### `steps`
+
+Information about the current operations Meilisearch is performing in this batch. A step may consist of multiple substeps.
+
+| Name              | Description                                        |
+| :-----------------| :------------------------------------------------- |
+| **`currentStep`** | A string describing the operation                  |
+| **`total`**       | The total number of operations in the step         |
+| **`finished`**    | The number of operations Meilisearch has completed |
+
+<Capsule intent="tip" title="">
+If Meilisearch is taking longer than expected to process a batch, monitor the `steps` array. If the `finished` field of the last item in the `steps` array does not update, Meilisearch may be stuck.
+</Capsule>
+
+#### `percentage`
+
+The percentage of completed operations, calculated from all current steps and substeps. This value is a rough estimate and may not always reflect the current state of the batch due to how different tasks are processed more quickly than others.
```

Comment:
Sorry, I'm bothering you again one last time on this part :confonded: 

```suggestion
The percentage of completed operations, calculated from all current steps and substeps. This value is a rough estimate and may not always reflect the current state of the batch due to how different steps are processed more quickly than others.
```
-----

And for more context, just so we're 100% aligned on the difference between a step and a task:

With Louis, we think we should **really** avoid using the term `task` in this context because we don't want the user to think that a batch is processing tasks in order or something.
We want to be sure that people don't misunderstand this:
- A batch is crafted by grouping multiple tasks together, that's true
- Once it starts processing there is no concept of task anymore, everything is merged together
- All the tasks are processing **together, at the same time**
=> That's why in this context, we really want to use the term « step » instead of task. We're processing the step « extracting word docids » for all the tasks at the same time.

Reply:
Thanks for this, it makes sense.

Just one question to clarify some aspects of the batching process. Is "once meilisearch starts processing the batch, there's no concept of task" a bit of an exaggeration? We still keep track of tasks so we can tell users task A was successful and task B failed, right?

Reply:
Yes, but in reality we know that beforehand.
Once we starts processing the batch, we don't have the information anymore